src/invitation.c

```diff
@@ -1196,6 +1196,10 @@ static bool invitation_receive(void *handle, uint8_t type, const void *msg, uint
 	return true;
 }
 
+static void invitation_rejected(void) {
+	fprintf(stderr, "Invitation rejected. Please consult server logs for details.\n");
+}
```

Comment:
You're right that the current (lack of) error message is bad. However, telling the person that does a `tinc join` to check server logs is not very helpful; they might not be the ones running the server. Maybe something like "contact the inviter for more details" might be better.

The other issue is that there are different ways to fail that even the client can distinguish. For example, I see you call this function also in case that it could not resolve the hostname in the invitation URL. That actually means the invitation was never used, so it isn't rejected. Only after sending the server the cookie can the invitation be rejected. If the connection is dropped before that, or if there is some name lookup problem, I think the error message should be "Could not contact the inviter."

Around line 1307, where we have `if (!ai)`, that's where we should explicitly give the error message "Could not resolve address from invitation URL", as `str2addrinfo()` will not print anything itself in this case.